plugins/optimization-detective/storage/data.php

```diff
@@ -85,9 +85,49 @@ function od_get_normalized_query_vars(): array {
 	return $normalized_query_vars;
 }
 
+/**
+ * Get the URL for the current request.
+ *
+ * This is essentially the REQUEST_URI prefixed by the scheme and host for the home URL.
+ * This is needed in particular due to subdirectory installs.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @return string Current URL.
+ */
+function od_get_current_url(): string {
```

Comment:
Inspired by the same function in the AMP plugin: https://github.com/ampproject/amp-wp/blob/5708e19c60cf0f88ce522349fba955266b6892f6/includes/amp-helper-functions.php#L617-L662